server/aesdsocket.c

```diff
@@ -0,0 +1,342 @@
+        }
+
+        /* change working directory to root */
+        int dirRes = chdir("/");
+        if(dirRes == -1)
+        {
+            syslog(LOG_ERR, "chdir failed, see errno for details");
+            printf("chdir failed, errno = %d", errno);
+            close(socketFD);
+            exit(-1);
+        }
+        
+        /* close std fd's as daemon cannot access terminal */
+        close(STDIN_FILENO);
+        close(STDOUT_FILENO);
+        close(STDERR_FILENO);
+    }
+
+    printf("\nlisten\n");
+/* -------------------------- listen for connections on created socket -------------------------- */
+    res = listen(socketFD, BACKLOG);
+    if(res != 0)
+    {
+        syslog(LOG_ERR, "listen failed, see errno for details");
+        printf("listen failed, see errno for details");
+        close(socketFD);
+        exit(-1);
+    }
+
+/* -------------------------- Loop for accepting connections and receiving packets -------------------------- */
+    while(!cleanExit)
+    {
+    /* accept incoming connection and save client info */
+        printf("\naccept\n");
+        clientSize = sizeof(clientAddr);
+        res = accept(socketFD, (struct sockaddr *)&clientAddr, &clientSize);
+        if(res == -1)
+        {
+            if(errno == EINTR)
+                continue;
+            syslog(LOG_ERR, "accept failed, see errno for details");
+            continue;
+        }
+    /* save connected client's FD for send/recv later */
+        clientFD = res;
+    /* log accepted connection and client's IP address */
+        inet_ntop(clientAddr.ss_family, get_in_addr((struct sockaddr *)&clientAddr), clientIP, sizeof(clientIP));
+        syslog(LOG_DEBUG, "Accepted connection from %s", clientIP);
+        printf("\nAccepted connection from %s", clientIP);
+
+    /* clear last recv data */
+        recvDone    = false;
+        totalBytes  = 0;
+        recvIndex   = 0;
+
+        while(!recvDone)
+        {
+        /* receive packet from connected client */
+            printf("\nrecv\n");
+            recvBytes = recv(clientFD, recvBuf, MAX_BUF_SIZE-1, 0);
+            if(recvBytes == -1)
+            {
+                if(errno == EINTR)
+                    continue;
+                else
+                {
+                    syslog(LOG_ERR, "recv failed, see errno for details");
+                    syslog(LOG_DEBUG, "closing connection from %s", clientIP);
+                    close(clientFD);
+                    exit(-1);
+                }
+                
+            }
+
+        /* track total bytes received */
+            totalBytes += recvBytes;
+        /* load recv buffer data into second buffer for file write */
+            for(i = 0; i < recvBytes; i++)
+                buf[recvIndex + i] = recvBuf[i];
+        
+        /* track buffer index from previous write */
+            recvIndex += recvBytes;
+        /* check for end of packet, \n terminated */
+            if(recvBuf[recvIndex - 1] == '\n')
```

Comment:
@mash9756 what would happen if `recv` returned multiple packets or portions of multiple packets, and `\n` wasn't the last byte in the buffer.  Are you guaranteed that `recv` will always return exactly one packet with end of packet at the last byte?